contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -0,0 +1,745 @@
+        Ownable2StepUpgradeable.acceptOwnership();
+    }
+
+    /**
+     * @notice Transfers the ownership of the contract to a new owner
+     * @param _newOwner Address of the new owner
+     * @dev Fixes solidity interface inference
+     * @dev Can only be called by the owner
+     */
+    function transferOwnership(address _newOwner) public override(IStakingVault, Ownable2StepUpgradeable) {
+        Ownable2StepUpgradeable.transferOwnership(_newOwner);
+    }
+
+    /**
+     * @notice Override the OwnableUpgradeable function to revert
+     */
+    function renounceOwnership() public view onlyOwner override(OwnableUpgradeable) {
+        revert RenouncementNotAllowed();
+    }
+
+    /**
+     * @notice Sets the depositor address
+     * @param _depositor Address of the new depositor
+     */
+    function setDepositor(address _depositor) external onlyOwner {
+        _setDepositor(_depositor);
+    }
+
+    /**
+     * @notice Ossifies the current implementation. WARNING: This operation is irreversible.
+     * @dev vault can't be connected to the hub after ossification
+     */
+    function ossify() external onlyOwner {
+        PinnedBeaconUtils.ossify();
+    }
+
+
+    /**
+     * @notice collects ERC20 tokens from the Staking Vault to the recipient
+     * @param _token Address of the token to recover
+     * @param _recipient Address of collection recipient
+     * @param _amount Amount of tokens to recover
+     */
+    function collectERC20(
+        address _token,
+        address _recipient,
+        uint256 _amount
+    ) external onlyOwner {
+        if (_token == address(0)) revert ZeroArgument("_token");
+        if (_recipient == address(0)) revert ZeroArgument("_recipient");
+        if (_amount == 0) revert ZeroArgument("_amount");
+        if (_token == RecoverTokens.ETH) {
+            revert EthCollectionNotAllowed();
+        }
+
+        RecoverTokens._recoverERC20(_token, _recipient, _amount);
+    }
+
+    /*
+     * ╔══════════════════════════════════════════════════╗
+     * ║ ┌──────────────────────────────────────────────┐ ║
+     * ║ │            INTERNAL FUNCTIONS                │ ║
+     * ║ └──────────────────────────────────────────────┘ ║
+     * ╚══════════════════════════════════════════════════╝
+     */
+
+    /**
+     * @dev Returns the storage struct for the ERC-7201 namespace
+     * @return $ storage struct for the ERC-7201 namespace
+     */
+    function _storage() private pure returns (Storage storage $) {
+        assembly {
+            $.slot := ERC7201_SLOT
+        }
+    }
+
+    function _depositToBeaconChain(Deposit calldata _deposit, bytes memory _withdrawalCredentials) internal {
+        uint256 balance = availableBalance();
+        if (_deposit.amount > balance) revert InsufficientBalance(balance, _deposit.amount);
+
+        DEPOSIT_CONTRACT.deposit{value: _deposit.amount}(
+            _deposit.pubkey,
+            _withdrawalCredentials,
```

Comment:
minor optimization: remove `_withdrawalCredentials` parameter and read WC directly in place, i.e. 
```solidity
DEPOSIT_CONTRACT.deposit{value: _deposit.amount}(
            _deposit.pubkey,
            bytes.concat(withdrawalCredentials()),
            ...
```